core/src/Streamly/Internal/Data/Array.hs

```diff
@@ -1238,17 +1257,19 @@ adaptCWith pstep initial extract cont !offset0 !usedCount !input = do
             -- IMPORTANT: the n here is from the byte stream parser, that means
             -- it is the backtrack element count and not the stream position
             -- index into the current input array.
-            ParserD.Done n b ->
-                assert (n >= 0)
-                    (cont (Success (- n) b) (count - n) None)
-            ParserD.Continue n pst1 ->
-                assert (n >= 0)
-                    (return $ Continue (- n) (parseCont (count - n) pst1))
+            ParserD.SDone m b ->
+                let n = 1 - m
```

Comment:
This looks incorrect based on the `IMPORTANT` note above. But why is no test failing?

Reply:
Was the `IMPORTANT` note incorrect?